src/core/server/csp/csp_config.test.ts

```diff
@@ -110,10 +110,63 @@ describe('CspConfig', () => {
         style_src: ['style'],
       });
       expect(config.header).toEqual(
-        `script-src 'unsafe-eval' 'self' script; worker-src blob: 'self' worker; style-src 'unsafe-inline' 'self' style`
+        `script-src 'self' script; worker-src blob: 'self' worker; style-src 'unsafe-inline' 'self' style`
       );
     });
 
+    describe('unsafe_eval', () => {
+      const isKibanaDistributable =
+        require('../../../../package.json').build?.distributable === true; // eslint-disable-line @typescript-eslint/no-var-requires
+
+      test('when "unsafe_eval" is set to `false`, the `unsafe-eval` CSP should not be set', () => {
+        const config = new CspConfig({
+          ...defaultConfig,
+          unsafe_eval: false,
+          script_src: ['foo', 'bar'],
+        });
+
+        expect(config.header).toEqual(
+          `script-src 'self' foo bar; worker-src blob: 'self'; style-src 'unsafe-inline' 'self'`
+        );
+      });
+
+      test('when "unsafe_eval" is set to `true`, the `unsafe-eval` CSP should be set', () => {
+        const config = new CspConfig({
+          ...defaultConfig,
+          unsafe_eval: true,
+          script_src: ['foo', 'bar'],
+        });
+
+        expect(config.header).toEqual(
+          `script-src 'self' 'unsafe-eval' foo bar; worker-src blob: 'self'; style-src 'unsafe-inline' 'self'`
+        );
+      });
+
+      if (isKibanaDistributable) {
+        test('in a distributable, when "unsafe_eval" is not set, the `unsafe-eval` CSP should be set', () => {
```

Comment:
> I think this is a code smell.

I agree

> Ihmo we should find a way to mock the `distributable` value to make it vary during tests instead of enabling these tests only in distributable/dev mode

I was thinking the same thing, but couldn't originally find a good way to mock it. Is gets it's default value in [`src/core/server/csp/config.ts`](https://github.com/elastic/kibana/pull/124484/files/711dc7c9f530964170749b0cdf6060b0eb0362ae#diff-bff7518a9e1005c9e619e3022f6e09118cf74ce719057e5ddfc492e64874c6f9). I guess maybe if I mock the exported `config.schema` object. I'll give that a try.

> when are unit tests ran in dist mode btw?

I assumed they are run in dist mode in CI, but maybe that's only functional/integration tests? If not, this implementation makes even less sense.

Reply:
> I assumed they are run in dist mode in CI, but maybe that's only functional/integration tests? If not, this implementation makes even less sense.

Yeah, it's also my understanding that unit tests don't run in `dist` mode. That only applies to functional tests IIRC

Reply:
Done 👍 I've created a mock that randomly flips the default value for `unsafe_eval` in this test